src/Compilers/Core/Portable/Compilation/RebuildData.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection.Metadata;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class RebuildData
+    {
+        /// <summary>
+        /// This represents the set of document names for the #line / #ExternalSource directives
+        /// that we need to emit into the PDB (in the order specified in the array).
+        /// </summary>
+        internal ImmutableArray<string> NonSourceFileDocumentNames { get; }
+
+        internal BlobReader OptionsBlobReader { get; }
+
+        internal RebuildData(
```

Comment:
>RebuildData [](http://example.com/codeflow?start=17&length=11)

Feels a bit odd to pass MetadataReader to the constructor since the object is not gonna hold on the reader (holding on an open MetadataReader has non-trivial lifetime management consequences). Maybe read the data in the caller and pass in the `ImmutableArray<string>`?

Reply:
Will do.